### DISCUSSION TITLE
Access custom attribute with Builder ( without ->get() )

### DISCUSSION THREAD

**Discussion by BenOussama180** [Power-Components/livewire-powergrid#753] [Q&A]
so basically im using powergrid tables livewire , in the datasource it expects a  Builder type : 
Datasource :
```
return Link::whereShowToCustomers(1)
            ->with('orders', 'orders.projects')
            ->orderBy('price', 'asc')
            ->groupBy('site');
```
in my  Link Model :
```
protected $appends = [
        'is_on_order',
    ];
public function getIsOnOrderAttribute()
    {
        return $this->orders->where('customer_id', auth()->id())->isEmpty();
    }
```
so basically i want to check if a link has been ordered by the current auth customer or no to use it inside my BooleanFilter : 
```
Column::add()
                ->title('Purchased Before')
                ->field('purchased_before_field', 'is_on_order')
                ->bodyAttribute(' text-center ', '')
                ->headerAttribute(' text-center ', '')
                ->makeBooleanFilter('is_on_order', 'No', 'Yes'),
```
### the issue is i can not access the "is_on_order" unless i perform a ->get() on my datasource which decrease performance of filtering alot , is there any workaround for this ?

**Comment by luanfreitasdev** [comment]
You need add in addColumn too

```php
 public function addColumns(): PowerGridEloquent
    {
        return PowerGrid::eloquent()
            ->addColumn('is_on_order')
```

**Reply by BenOussama180** [reply]
i already did that , still get the error while filtering , unkown column in database 'is_on_order' , i only get rid of that eerror when i perform a '->get() ' inside my datasource but its not  a good practice

**Reply by BenOussama180** [reply]
![image](https://user-images.githubusercontent.com/81111919/206552083-caebc759-65ba-4e65-b62f-08c6c06a480c.png)
![image](https://user-images.githubusercontent.com/81111919/206552122-52db2167-d1d1-4373-863d-08d64907815e.png)
![image](https://user-images.githubusercontent.com/81111919/206552157-5853b166-a533-4c32-8923-fa34eb1dce1c.png)
